### ISSUE TITLE
File Watchers are triggered on opening files

### ISSUE THREAD

**Issue by shelbykauth** [microsoft/vscode#112102]
Issue Type: <b>Bug</b>

I see lots of bugs about file watchers not catching changes.  But this is about file watchers catching too many changes.

- npm scripts running in integrated terminal:
  ```
    "start": "nodemon --exec babel-node ./src/index.js",
    "test:dev": "mocha --watch --reporter min",
  ```
- Be on Windows.  Don't be on MacOS (I tested on Mac, doesn't happen)
- Have lots of files opened.  Possibly depends upon using up most of the Ram (I have 6.5GB in use, 1.3GB Available/Cached, if I close out all my programs, I still have most of my memory in use.  That's a different story, not the focus, and possibly irrelevant.)
- When clicking on different file tabs, and switching between Source Control action button and Explorer action button, and sometimes as I type (not saving), and usually as I stage Git changes (again, not saving), mocha and nodemon will restart.
- It doesn't happen all the time.  Re-clicking on the same file does not show the bug.  It has to be one I haven't viewed in a while.  Opening a file that was not open recently also does not trigger it.
  - It's possible that it had to be saved recently, then tabbed out of, then tabbed back into.  I'm not sure.

Expected Behavior: File Watchers should restart when I manually save the watched files, or one of my programs writes to a file/directory that is under the file watcher.  File Watchers should not restart when opening a file, staging changes, closing a file, etc.

VS Code version: Code 1.51.1 (e5a624b788d92b8d34d1392e4c4d9789406efe8f, 2020-11-10T23:34:32.027Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8350U CPU @ 1.70GHz (8 x 1896)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.85GB (1.04GB free)|
|Process Argv|--crash-reporter-id ff713049-a5b1-4539-9c83-593f7ba49a85|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none

**Comment by shelbykauth** [comment]
Update:
<details>
<summary>
I am able to trigger it more consistently by the following steps:

  - Have a file with saved, uncommitted changes.
  - Save file (no unsaved changes needed) (ctrl+s)
    - Watcher triggers.  This is good.
  - Stage changes (via UI in SourceControl using the plus button on the file)
    - Watcher triggers.  This is not good.
</summary>

  - Unstage changes (via UI using the minus button)
    - Watcher sometimes triggers.  It should not trigger
  - Stage changes (via UI)
    - Watcher does not trigger.  This is good
  - Repeating stage/unstage does not trigger watcher
  - Save
    - Watcher triggers.  This is good.
  - Unstage
    - Watcher triggers.  This is not good.

</details>

In short, save the file.  The next stage/unstage after save triggers the watchers consistently when it shouldn't.

Still working on figuring out what triggers it when I open or go back to files.

**Comment by bpasero** [comment]
https://github.com/microsoft/vscode/issues/112102#issuecomment-740819999 sounds like git causing external watchers to fire. Unrelated to VSCode's built in watcher.

**Comment by joaomoreno** [comment]
> The next stage/unstage after save triggers the watchers consistently when it shouldn't.

When git executes, it will mess with the `.git` folder. Are you ignoring that folder in your nodemon/mocha watcher? This issue isn't about VS Code at all, but about git.
